### PR TITLE
Use graphql-core-next instead of graphql-core + support async

### PR DESCRIPTION
**Pull request by norman-thomas**
I made some adaptions to use this library together with [graphql-core-next](https://github.com/graphql-python/graphql-core-next)

The usage of `async` / `await` may not be desired in this repo for reasons that start with `2.x`. In that case, feel free to close/ignore.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from setuptools import setup, find_packages
 
-required_packages = ["graphql-core>=2.1", "promise"]
+required_packages = ["graphql-core-next>=1.0.2", "promise"]
```
Comment (dan98765) [inline setup.py]: Just a question: how did you pick this version to `>=` for?

File: setup.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
     keywords="api graphql protocol rest",
     packages=find_packages(exclude=["tests"]),
     install_requires=required_packages,
-    tests_require=["pytest>=2.7.3"],
+    tests_require=["pytest>=4.3.1"],
```
Comment (dan98765) [inline setup.py]: I don't see in the CR where you're changing or adding tests that use any new features of pytest; why did you bump the version here?

**Comment by Cito** [comment]
Thank you @norman-thomas 

Unfortunately, I saw it only today - I have also created a branch of graphql-server-core yesterday [here](https://github.com/Cito/graphql-server-core/tree/use-core-next) that has similar changes.

graphql-server-core is actually a helper library for other libraries like flask-graphql. If we make any changes here, we must make sure that these libraries do not get broken or upgrade them as well.

In my branch I added tests so that we have 100% code coverage. But some things are still different due to the new API, and tools like flask-grapqhl will need some adaptation, too.

Before we proceed I think we need a general versioning policy for the graphql-python tools so that the full stack of tools stays compatible (see graphql-python/graphql-core/issues/241 for that discussion).

**Comment by KingDarBoja** [comment]
Closing as fixed on #36

**Pull request closed without merging**